network/p2p/peerManager.go

```diff
@@ -10,36 +10,29 @@ import (
 
 	"github.com/onflow/flow-go/engine"
 	"github.com/onflow/flow-go/model/flow"
-	"github.com/onflow/flow-go/model/flow/filter"
 )
 
 // Connector connects to peer and disconnects from peer using the underlying networking library
 type Connector interface {
 
-	// ConnectPeers connect to the given flow.Identities and returns a map of identifiers which failed.
-	// ConnectPeer implementation should be idempotent such that multiple calls to connect to the same peer should not
+	// UpdatePeers connects to the given flow.Identities and returns a map of identifiers which failed. It also
```

Comment:
changed the code to return the `UnconvertableIdentitiesError` error. However, the best that can be done is to `log.fatal` when that error occurs since the discovery (connection/disconnection) is all async and happens on-request.